src/Bab/RabbitMq/VhostManager.php

```diff
@@ -188,17 +188,22 @@ private function createQueues(Configuration $config)
                     $this->createBinding('retry', $name, $name);
                 }
 
-                $retryName = $name.'_retry_'.($i+1);
+                $retryQueueName = $name.'_retry_'.$retries[$i];
+                $queues = $this->getQueues();
```

Comment:
Instead of looking at all queues, I think you should build a list of all retry queues to create for the giving queue.